core/string/ustring.cpp

```diff
@@ -3655,6 +3655,30 @@ bool String::is_absolute_path() const {
 	}
 }
 
+static _FORCE_INLINE_ bool _is_valid_identifier(int p_index, char32_t p_char) {
```

Comment:
This needs a better name, because it only handles a single char of an identifier.
Maybe something like `is_valid_identifier_bit()`.